src/main/java/org/jabref/logic/exporter/AtomicFileOutputStream.java

```diff
@@ -0,0 +1,197 @@
+ * kept.</li>
+ * </ol>
+ *
+ * Implementation inspired by code from <a href="https://github.com/martylamb/atomicfileoutputstream/blob/master/src/main/java/com/martiansoftware/io/AtomicFileOutputStream.java">Marty
+ * Lamb</a> and <a href="https://github.com/apache/zookeeper/blob/master/src/java/main/org/apache/zookeeper/common/AtomicFileOutputStream.java">Apache</a>.
+ */
+public class AtomicFileOutputStream extends FilterOutputStream {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AtomicFileOutputStream.class);
+
+    private static final String TEMPORARY_EXTENSION = ".tmp";
+    private static final String BACKUP_EXTENSION = ".sav";
+
+    /**
+     * The file we want to create/replace.
+     */
+    private final Path targetFile;
+    /**
+     * The file to which writes are redirected to.
+     */
+    private final Path temporaryFile;
+    /**
+     * A backup of the target file (if it exists), created when the stream is closed
+     */
+    private final Path backupFile;
+
+    /**
+     * Creates a new output stream to write to or replace the file at the specified path.
+     *
+     * @param path the path of the file to write to or replace
+     */
+    public AtomicFileOutputStream(Path path) throws IOException {
+        super(Files.newOutputStream(getPathOfTemporaryFile(path)));
+
+        targetFile = path;
+        temporaryFile = getPathOfTemporaryFile(path);
+        backupFile = getPathOfBackupFile(path);
+    }
+
+    private static Path getPathOfTemporaryFile(Path targetFile) {
+        return FileUtil.addExtension(targetFile, TEMPORARY_EXTENSION);
+    }
+
+    private static Path getPathOfBackupFile(Path targetFile) {
+        return FileUtil.addExtension(targetFile, BACKUP_EXTENSION);
+    }
+
+    /**
+     * Returns the path of the backup copy of the original file (may not exist)
+     */
+    public Path getBackup() {
+        return backupFile;
+    }
+
+    /**
+     * Override for performance reasons.
+     */
+    @Override
+    public void write(byte b[], int off, int len) throws IOException {
+        try {
+            out.write(b, off, len);
+        } catch (IOException exception) {
+            cleanup();
+            throw exception;
+        }
+    }
+
+    /**
+     * Closes the write process to the temporary file but does not commit to the target file.
+     */
+    public void abort() throws IOException {
+        try {
+            super.close();
+            Files.deleteIfExists(temporaryFile);
+            Files.deleteIfExists(backupFile);
+        } catch (IOException exception) {
+            LOGGER.debug("Unable to abort writing to file " + temporaryFile, exception);
+        }
+    }
+
+    private void cleanup() {
+        try {
+            Files.deleteIfExists(temporaryFile);
+        } catch (IOException exception) {
+            LOGGER.debug("Unable to delete file " + temporaryFile, exception);
```

Comment:
For arguments in the logger you can simply use a pair of curly braces {} instead of concat with plus. 
https://www.slf4j.org/faq.html#logging_performance
Here and in above

Reply:
This is not possible here since the there is no appropriate debug method that also accepts an exception.